### DISCUSSION TITLE
Populating 1 bar parts to 4 bar patterns?

### DISCUSSION THREAD

**Discussion by ptowns** [bangcorrupt/hacktribe#47] [Ideas]
Is it possible to copy a 1 bar part to a 4 bar pattern and have it fill out all 4 bars?
The work around seems to be set pattern to 1 bar, copy part then make pattern 4 bars. BUT this only seems to work once. If pattern was 4 and you change it to 1 then back again it dosnt seem to work for me???

**Comment by BKLronin** [comment]
I often start with one bar, safe and change length to 4 and it's copied across all bars ?

**Comment by bangcorrupt** [comment]
>  start with one bar, safe and change length to 4 and it's copied across all bars 

> if pattern was 4 and you change it to 1 then back again it dosnt seem to work

I think it works like this to prevent losing data when the pattern length is reduced.  It would be good if it always copied only if all the next bars are empty.  

For now, this could be done externally using SysEx to Read-Modify-Write the pattern file.  Keep an eye on [e2-scripts/#3](https://github.com/bangcorrupt/e2-scripts/issues/3).

**Comment by bangcorrupt** [comment]
Try this:

1. In sequencer mode, clear bars 2-4 of the part you want to copy.
2. Set pattern length to 1 and write pattern.
3. Change pattern and initialize new 1 bar pattern.
4. Copy part from first pattern to new pattern.
6. Set pattern length to 4 and write pattern.
7. Copy part back to first pattern.

**Reply by ptowns** [reply]
So make my 1 bar parts 4 bar.... that should work!   Thank you! This could be huge for me. Answer was right under my nose!
